packages/core/src/components/ConfirmationModal/ConfirmationModal.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { noop } from '../../common/helpers';
+import { Modal, Button } from '../../index';
+
+const ConfirmationModal = ({
+  show,
+  onCancel,
+  onConfirm,
+  title,
+  icon,
+  heading,
+  body,
+  cancelButtonText,
+  confirmButtonText,
+  className,
+  footer,
+  ...otherProps
+}) => (
+  <Modal className={className} show={show} onHide={onCancel} {...otherProps}>
+    <Modal.Header>
+      <Modal.CloseButton onClick={onCancel} />
+      <Modal.Title>{title}</Modal.Title>
+    </Modal.Header>
+    <Modal.Body>
+      <div className="pf-confirm-modal-body">
+        {icon && <span className="pf-confirm-modal-icon">{icon}</span>}
+        <span className="pf-confirm-modal-content">
+          <span className="spacer" />
+          {heading && (
+            <div className="pf-confirm-modal-content-heading">{heading}</div>
+          )}
+          {body && <div className="pf-confirm-modal-content-body">{body}</div>}
+          <span className="spacer" />
+        </span>
+      </div>
+    </Modal.Body>
+    <Modal.Footer>
+      {!footer ? (
+        <React.Fragment>
+          <Button bsStyle="default" className="btn-cancel" onClick={onCancel}>
+            {cancelButtonText}
+          </Button>
+          <Button autoFocus bsStyle="primary" onClick={onConfirm}>
+            {confirmButtonText}
+          </Button>
+        </React.Fragment>
+      ) : (
+        footer
+      )}
+    </Modal.Footer>
+  </Modal>
+);
+
+ConfirmationModal.propTypes = {
+  /** class names */
+  className: PropTypes.string,
+  /** show/hide modal */
+  show: PropTypes.bool.isRequired,
+  /** callback to trigger when closing the modal or clicking the default footer cancel button */
+  onCancel: PropTypes.func.isRequired,
+  /** callback to trigger when clicking the default footer confirm button */
+  onConfirm: PropTypes.func,
+  /** modal title */
+  title: PropTypes.string,
+  /** modal body icon */
+  icon: PropTypes.node,
+  /** modal body heading */
+  heading: PropTypes.node,
+  /** modal body */
+  body: PropTypes.node,
+  /** text for default footer cancel button */
+  cancelButtonText: PropTypes.string,
```

Comment:
Since this is really just a strict abstraction around Modal - it doesn't seem that important I suppose. A user could still extend this w/ different button variations like you suggest.

Reply:
I'm not very familiar with this design pattern. Maybe @serenamarie125 or @mcarrano can weigh in on whether the footer buttons could ever include anything other than text.

Reply:
We are currently in the process of introducing a confirmation dialog pattern to the design library. You can follow that issue here: https://github.com/patternfly/patternfly-design/issues/632. But in short, if I understand your question @jgiardino , we have no plan to use anything other than text buttons as confirmation actions on the dialog.

Reply:
I think it'd be ok to change to `PropTypes.Node` for the off chance that an application needs to include an icon in the button. I don't feel we need to be restrictive.